### DISCUSSION TITLE
How does this install on MacOS?

### DISCUSSION THREAD

**Discussion by mandyyli** [PhilipK/BoilR#229] [Q&A]
I am trying to install BoilR on a M1 Macbook. I downloaded the macos installer but it doesn't have an extension. When opening, it just opens as a document. Anyone know what I'm doing wrong?

I'll also add that I tried manually adding .pkg and .dmg extension, but both result in an error.

**Comment by PhilipK** [comment]
What is the usual extension on M1? I am sorry, I don't know.

**Reply by mandyyli** [reply]
It is usually .dmg for the disc installers or .pkg which uses MacOS Installer. What type of installer is the BoilR on source release? I'm not sure why there isn't a file extension there by default.

**Reply by PhilipK** [reply]
It is not an installer, it is just a standalone executable.
I do not have access to a M1 Macbook, so I just told the compiler to output a M1 executable, and have honestly not been able to test it myself.

**Reply by mandyyli** [reply]
Ah I see. I tried to run it as an executable but gives me a `zsh: killed` error. Google tells me this is due to a codesigning error - M1 is strict with running apps unless they have been codesigned by the developer. I tried manually codesigning locally using `codesign -s` but still doesn't work. 

I don't know much about M1 app development so have no clue where to go next. Thank you for the help anyway!
